Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -851,12 +866,23 @@ extension PackageInfo {
 }
 
 extension PackageInfo.Target {
-    var relativePath: RelativePath {
-        RelativePath(path ?? "Sources/\(name)")
+    func mainPath(packageFolder: AbsolutePath) throws -> AbsolutePath {
+        if let path = path {
+            return packageFolder.appending(RelativePath(path))
+        } else {
+            let firstMatchingPath = PackageInfoMapper.predefinedSourceDirectories
```

Comment:
I'd add comment here or at `predefinedSourceDirectories` that details the SPM behavior we have observed and that has made us do this change as it might not be obvious in retrospect.

Reply:
Could also link to this file: https://github.com/apple/swift-package-manager/blob/751f0b2a00276be2c21c074f4b21d952eaabb93b/Sources/PackageLoading/PackageBuilder.swift#L488